cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -72,35 +71,15 @@ public CruiseControlApiImpl(Vertx vertx) {
     }
 
     @Override
-    public Future<Boolean> isProposalReady(String host, int port) {
-        return isProposalReady(host, port, null);
-    }
-
-    public Future<Boolean> isProposalReady(String host, int port, String userTaskID) {
-        return getCruiseControlState(host, port, false, userTaskID).compose(stateResponse -> {
-            Boolean proposalReady = stateResponse
-                            .getJson()
-                            .getJsonObject("AnalyzerState")
-                            .getBoolean("isProposalReady");
-            return Future.succeededFuture(proposalReady);
-        });
-
-    }
-
-    @Override
-    public Future<CruiseControlResponse> rebalance(String host, int port, RebalanceOptions rbOptions) {
-        return rebalance(host, port, rbOptions, null);
-    }
-
     @SuppressWarnings("deprecation")
-    public Future<CruiseControlResponse> rebalance(String host, int port, RebalanceOptions rbOptions, String userTaskId) {
+    public Future<CruiseControlRebalanceResponse> rebalance(String host, int port, RebalanceOptions rbOptions, String userTaskId) {
```

Comment:
so now we have the only the `rebalance` method which gets the `userTaskId`, can you explain what's the role of `userTaskId` in a rebalance request? I see it's put in the header of the request but it's not documented on the cruise control wiki. Is that for doing the actual rebalance of a specific rebalance proposal identified via this id?

Reply:
I think it is used to track the status of a  request [1]

[1] https://github.com/linkedin/cruise-control/blob/a9b6b4262130ef488e59599e1f2e0198ed7c3c97/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java#L59-L64

Reply:
Kyle is right, the user task ID is need in the case where the optimization requests via the `rebalance` endpoint took longer than the configured time out to complete. In this case a "in progress" response is returned and we need to keep hold of the user task ID to repeat the request in the next reconciliation. This applies only to the `dry=true` case as in the `dryrun=false` we poll the `user_tasks` end point instead.